napari/_tests/test_adding_removing.py

```diff
@@ -138,3 +138,23 @@ def my_custom_callback():
         # warningEmitters are not connected when connecting to the emitterGroup
         if not isinstance(em, WarningEmitter):
             assert len(em.callbacks) == 1
+
+
+def test_adding_with_contrast_limits(make_napari_viewer):
+    viewer = make_napari_viewer()
+    image = np.asarray([[0, 1], [np.nan, np.inf]])
+    viewer.add_image(image, contrast_limits=[0, 1])
+
+    num_errors = 0
+    try:
+        viewer.add_image(image, contrast_limits=[1, 1])
+    except TypeError:
+        num_errors = num_errors + 1
+
+    try:
+        viewer.add_image(image, contrast_limits=[1, 0])
+    except TypeError:
+        num_errors = num_errors + 1
+
+    if num_errors != 2:
+        assert False
```

Comment:
would be much faster (and not require Qt) to test this without creating a full viewer.  Also, use `assert pytest.raises()` rather than try/except then assert False.

```python
def test_contrast_limits_must_be_increasing():
    with pytest.raises(ValueError):
        Image(np.random.rand(8, 8), contrast_limits=[1, 1])
    with pytest.raises(ValueError):
        Image(np.random.rand(8, 8), contrast_limits=[1, 0])
```

Reply:
... and then move to one of the tests in the layers folder

Reply:
👍